File: 2 term/PB/PB Lab2.5/PB Lab2.5 cs/Prism4.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PB_Lab2._5_cs
{
    internal class Prism4 : Prism
    {
        public Prism4(double height, double sideLength) : base(height, sideLength)
        {}

        public override double CalculateBaseSurface()
        {
            return SideLength * SideLength;
        }

        public override double CalculateSurface()
        {
            return CalculateBaseSurface() * 2 + SideLength * Height * 4;
        }

        public override double CalculateVolume()
        {
            return CalculateBaseSurface() * Height;
        }
    }
}

File: 2 term/PB/PB Lab2.5/PB Lab2.5 py/Prism4.py
from Prism import Prism


class Prism4(Prism):
    def calculate_base_surface(self):
        return self.side_length ** 2

    def calculate_surface(self):
        return 2 * self.calculate_base_surface() + 4 * self.side_length * self.height

